src/components/DateInput/DateInput.stories.mdx

```diff
@@ -0,0 +1,135 @@
+3. [TextInput](http://localhost:6006/?path=/docs/components-form-inputs-textinput--default-story)
+
+The `DateInput` uses sensible defaults for these to allow for very straightforward ergonomics that allow users to
+select dates with minimal work.
+
+<Canvas>
+  <Story name="default">
+    {() => {
+      const [selectedDate, setSelectedDate] = useState(null);
+      const handleClear = () => {
+        setSelectedDate(null);
+      };
+      return (
+        <Box childGap="md">
+          <Box height="300px" width="300px">
+            <DateInput
+              datePickerProps={{
+                selected: selectedDate,
+                onChange: setSelectedDate,
+              }}
+              textInputProps={{
+                placeholder: '11/02/2020',
+                onClear: handleClear,
+                id: 'myDatePicker',
+                name: 'selectDate',
+                label: 'Select Date',
+              }}
+            />
+          </Box>
+        </Box>
+      );
+    }}
+  </Story>
+</Canvas>
+
+## Date Range
+
+<Canvas>
+  <Story name="range">
+    {() => {
+      const [startDate, setStartDate] = useState(null);
+      const [endDate, setEndDate] = useState(null);
+      const setDate = ([startDate, endDate]) => {
+        setStartDate(startDate);
+        setEndDate(endDate);
+      };
+      const handleClear = () => {
+        setStartDate(null);
+        setEndDate(null);
+      };
+      return (
+        <Box childGap="md">
+          <Box height="300px" width="500px">
+            <DateInput
+              datePickerProps={{
+                onChange: setDate,
+                selected: startDate,
+                selectsRange: true,
+                startDate,
+                endDate,
+              }}
+              textInputProps={{
+                placeholder: '11/02/2020',
+                onClear: handleClear,
+              }}
+            />
+          </Box>
+        </Box>
+      );
+    }}
+  </Story>
+</Canvas>
+
+## With custom date format
+
+You can manipulate the format on the input field by using the `dateFormat` and `dateOptions`
+props. Under the hood our component uses the [date-fns `format` function](https://date-fns.org/v2.16.1/docs/format)
+to generate a string based on these options.
+
+<Canvas>
+  <Story name="Custom date format">
+    {() => {
+      const [selectedDate, setSelectedDate] = useState(null);
+      const [inputValue, setInputValue] = useState('');
+      const handleClear = () => {
+        setSelectedDate(null);
+      };
+      return (
+        <Box childGap="md">
+          <Box height="300px" width="300px">
+            <DateInput
+              dateFormat={'M/d/yy'}
+              datePickerProps={{
+                selected: selectedDate,
+                onChange: setSelectedDate,
+              }}
+              textInputProps={{
+                placeholder: '11/02/20',
+                onClear: handleClear,
+                id: 'myDatePicker',
```

Comment:
```suggestion
                id: 'myCustomDatePicker',
```

Make `id` unique so we don't have a bunch of console warnings in our documentation.